src/NITTA/Model/Problems/Refactor/OptimizeLut.hs

```diff
@@ -0,0 +1,214 @@
+-- All extensions should be enabled explicitly due to doctest in this module.
```

Comment:
Good point, but I don't find a doctest in the file.

Looks like it will be nice if you illustrate how refactoring applies in the doctest.